packages/block-library/src/template-part/edit/import-controls.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo, useState } from '@wordpress/element';
+import { useDispatch, useSelect, useRegistry } from '@wordpress/data';
+import {
+	Button,
+	Flex,
+	FlexBlock,
+	FlexItem,
+	SelectControl,
+} from '@wordpress/components';
+import { switchToBlockType } from '@wordpress/blocks';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { useCreateTemplatePartFromBlocks } from './utils/hooks';
+import { transformWidgetToBlock } from './utils/transformers';
+
+const WILDCARD_TRANSFORMS = [ 'core/columns', 'core/group' ];
+
+export function TemplatePartImportControls( { area, setAttributes } ) {
+	const [ selectedSidebar, setSelectedSidebar ] = useState( '' );
+	const [ isBusy, setIsBusy ] = useState( false );
+
+	const registry = useRegistry();
+	const sidebars = useSelect( ( select ) => {
+		return select( coreStore ).getSidebars( {
+			per_page: -1,
+			_fields: 'id,name,description,status,widgets',
+		} );
+	}, [] );
+	const { getBlockTransformItems } = useSelect( blockEditorStore );
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	const createFromBlocks = useCreateTemplatePartFromBlocks(
+		area,
+		setAttributes
+	);
+
+	const options = useMemo( () => {
+		const sidebarOptions = ( sidebars ?? [] )
+			.filter( ( widgetArea ) => widgetArea.widgets.length > 0 )
+			.map( ( widgetArea ) => {
+				return {
+					value: widgetArea.id,
+					label: widgetArea.name,
+				};
+			} );
+
+		if ( ! sidebarOptions.length ) {
+			return [];
+		}
+
+		return [
+			{ value: '', label: __( 'Select widget area' ) },
+			...sidebarOptions,
+		];
+	}, [ sidebars ] );
+
+	async function createFromWidgets( event ) {
+		event.preventDefault();
+
+		if ( isBusy || ! selectedSidebar ) {
+			return;
+		}
+
+		setIsBusy( true );
+
+		const sidebar = options.find(
+			( { value } ) => value === selectedSidebar
+		);
+		const { getWidgets } = registry.resolveSelect( coreStore );
+
+		// The widgets API always returns a successful response.
+		const widgets = await getWidgets( {
+			sidebar: sidebar.value,
+			_embed: 'about',
+		} );
+
+		const skippedWidgets = new Set();
+		const blocks = widgets.flatMap( ( widget ) => {
+			const block = transformWidgetToBlock( widget );
+
+			if ( block.name !== 'core/legacy-widget' ) {
+				return block;
+			}
+
+			const transforms = getBlockTransformItems( block ).filter(
+				( item ) => ! WILDCARD_TRANSFORMS.includes( item.name )
```

Comment:
I'm unsure if there's a better way to skip wildcard transformations.

Reply:
I played around with the `wp.blocks` in the DevTools console and here's the best I could come up with:

<img width="731" alt="Screenshot 2022-12-02 at 16 17 11" src="https://user-images.githubusercontent.com/612155/205220135-d10d0c8b-9e20-4128-bd1b-33758d2f4f91.png">

I'd prefer to go with something along those lines because future developers are definitely not going to remember to update `WILDCARD_TRANSFORMS` here when a new wildcard transform is introduced.

Reply:
I believe the usage of `getBlockTransformItems` is preferred over the `blocks` API methods. But I agree we should use a more general solution to skip wildcard transformations. 

@ntsekouras, do you have any recommendations on the best way to achieve this?

Reply:
Transforms API would need some improvement especially the wildcard character yes, but it's out of scope here. I don't have a strong opinion here, but we make sure not to use hardcoded block names. Whatever is simpler it could be okay I think.

What is the reason we want to skip the wildcard ones though? Can we skip this check altogether.

Reply:
> What is the reason we want to skip the wildcard ones though? Can we skip this check altogether.

We're trying to transform Legacy Widgets during the import, and the only available transformations for this block are either actual transformation or wildcards. So removing the wildcard transformations gives us the correct block for converting.

Reply:
Updated in [f42321e](https://github.com/WordPress/gutenberg/pull/45509/commits/f42321e2df2a288a434c2b9a8510bddc1796eee3).